python/ray/util/dask/scheduler.py

```diff
@@ -21,6 +21,13 @@
 pools = defaultdict(dict)
 pools_lock = threading.Lock()
 
+DASK_TO_RAY_RESOURCES = [
+    ("CPU", "num_cpus"),
+    ("GPU", "num_gpus"),
+    ("memory", "memory"),
+    ("object_store_memory", "object_store_memory"),
```

Comment:
Dask knows about object store?

Reply:
Dask does not know about the object store; this map can be thought of as "resources to pluck out of `resources` and give as top-level Ray task options under a new name", i.e. these are resources that we do not allow to be in the `ray.remote(resources={...})` argument and that _have_ to be given as top-level. Dask Distributed users are going to be used to giving their CPU and GPU resource requests as part of the `resources={...}` dict, either to `.compute()` or as an `.annotate()` annotation, so this mapping makes this transition a bit easier.

The other option is to let Ray throw an error when these resources are given in the `ray.remote(resources={...})` argument and make the user give `num_cpus`, `num_gpus`, etc., as top-level `.compute()` and `.annotate()` arguments. I'd like to give the user the option to keep all resource requests in the `resources={...}` dict since I think that's a nice, clean option, but happy to keep this more in line with our `ray.remote()` API.

Reply:
I think in general I'd prefer only allowing `ray_remote_args`, instead of supporting both. As a user having both available is somewhat confusing, and I worry about one being preferable over the other. If we only supported `ray_remote_args`, and had DASK->Ray translation in the docs, the ambiguity goes away.

Reply:
> The other option is to let Ray throw an error when these resources are given in the ray.remote(resources={...}) argument.

Upon further thought I like this option. Can we do this, and also update the docs to make it clear `resources` is for Dask compatibility only? Users can just use ray_remote_args if they don't need compat.

Reply:
I actually agree with your initial point. I'm thinking that it'd be better to either completely accommodate the top-level `resources={...}` API (as currently exists in the PR), or only accept `ray_remote_args`. Only supporting the `resources={...}` API for custom (non-system) resources seems like bad UX.

```python
# (1) Current PR
with dask.annotate(resources={"CPU": 1, "foo": 1}):
    col = ...

# (2) Custom resources + remote args
with dask.annotate(resources={"foo": 1}, ray_remote_args={"num_cpus": 1}):
    col = ...

# (3) Remote args only
with dask.annotate(ray_remote_args={"num_cpus": 1, "resources": {"foo: 1}}}):
    col = ...
```

Reply:
Hmm, then maybe we should go for (3), which optimizes for clarity over perfect compatibility. In that case, we can raise an warning/error if "resources" is found in the kwargs, telling the user to use ray_remote_args.

Reply:
Agreed! I'll make that change.

Reply:
Change is made. Compared to allowing a top-level `resources` along with `num_cpus`, `num_gpus`, etc., I feel as if this is slightly worse UX, see example: https://github.com/ray-project/ray/pull/22057/commits/6e8a8fc14369f2a4d68da817b2efbf216c5cb420#diff-cc2090d8130a6fee7ad03b26c646dc1640107533f4e05f6cca02eb7d8933553e